### ISSUE TITLE
calcite dropdown cannot be navigated with keyboard when containing a disabled item

### ISSUE THREAD

**Issue by janett-baresel** [Esri/calcite-components#4944]
### Summary

When creating a dropdown with a disabled dropdown item, the user cannot reach any elements passed the disabled item.

### Actual Behavior

When tabbing through the dropdown items the dropdown closes when the focus would reach the disabled item.

Note: if I use a dropdown group with the same items, this works as expected.

### Expected Behavior

When tabbing through the dropdown items the dropdown stays open and the focus skips the disabled dropdown item and goes to the next item. (This works as expected for item sin a dropdown group)

### Reproduction Sample

https://codepen.io/Programmine/pen/jOzVvRN?editors=100

### Reproduction Steps

1. create `calcite-dropdown` without `calcite-dropdown-group` with 3 items
2. make the middle item disabled
3. use your keyboard to navigate passed the disabled item
-> dropdown closes and focus moves on somewhere else

### Reproduction Version

latest (coming from codepan) so I assume beta86?

<SUP>SVCH</SUP>

**Comment by driskull** [comment]
@benelan @jcfranco I think dropdown should be navigated via arrow keys instead of using tab. Tab should probably skip out of the dropdown to the next focusable element in the DOM.

Example: https://www.w3.org/WAI/ARIA/apg/patterns/menubutton/
https://www.w3.org/WAI/ARIA/apg/example-index/menu-button/menu-button-actions-active-descendant.html

@geospatialem Is that correct?

Maybe we can use this issue to update it to use arrow keys instead of tab?

**Comment by benelan** [comment]
> I think dropdown should be navigated via arrow keys instead of using tab. Tab should probably skip out of the dropdown to the next focusable element in the DOM.

That sound intuitive to me, but I'll defer to kitty/franco

**Comment by geospatialem** [comment]
> Tab should probably skip out of the dropdown to the next focusable element in the DOM.
> @geospatialem Is that correct?

@driskull Yes, that is the expected behavior, using arrow keys to access the items. 👍🏻 

This only seems to apply when the `dropdown-group` isn't present, as Janett mentioned, so the initial behavior when using `dropdown-group` is performing as intended. Should we account for the missing group? If there is no specified `groupTitle` prop on the group, only the items are shown, like this [example](https://codepen.io/geospatialem/pen/eYMgYbb).

Also, it doesn't seem we have a `disabled` prop for [`dropdown-item`](https://developers.arcgis.com/calcite-design-system/components/dropdown-item/#component-api-properties)?

**Comment by driskull** [comment]
We may want to document that all dropdown items are contained within a `calcite-dropdown-group` and that its mandatory? The group [sets the menu role](https://github.com/Esri/calcite-components/blob/master/src/components/dropdown-group/dropdown-group.tsx#L86). cc @macandcheese @benelan @jcfranco 

Should we throw an error or something if items are not within a group?

**Comment by benelan** [comment]
> Should we throw an error or something if items are not within a group?

I'd say so, since in other components groups are optional so people may assume the same for dropdown.

**Comment by geospatialem** [comment]
> We may want to document that all dropdown items are contained within a `calcite-dropdown-group` and that its mandatory? The group [sets the menu role](https://github.com/Esri/calcite-components/blob/master/src/components/dropdown-group/dropdown-group.tsx#L86).

Good idea. Will update the docs to include this. 👍🏻

**Comment by driskull** [comment]
It might be best to actually refactor the roles so that the menu role is on the dropdown instead of the dropdown-group. The dropdown-group should have a "group" role. Then we can just make keyboard navigation work without a `group`. Thoughts?

**Comment by macandcheese** [comment]
Multiple groups with mixed selection modes can be contained within a single dropdown, if they are all the same role moving it to the parent is fine, but we should make sure they don’t each need a unique role.

**Comment by macandcheese** [comment]
If we make groupTitle required we’d need a way to prevent its display (optionally) - there are times it may not make sense to display the groupTitle (the dividing line still appears) but have it read.

For the mixed groups, I meant that we should ensure the roles do not need to be different internally, nothing to do with documentation for users. 

Basically, do dropdown group of selection mode single / multiple need to be listbox? In the ul / li example above, that is an example of the “selection mode none” group.

**Comment by jcfranco** [comment]
I noticed a difference in default selection behavior when you have a group (`selection-mode: single`) vs without (`selection-mode: none` or behaves like it): https://codepen.io/jcfranco/pen/PoRKdgz?editors=1000

Seems confusing to have them behave differently.

**Comment by driskull** [comment]
Looks like a separate issue. @jcfranco can you create an issue for that? 

Looks like the dropdown item expects a `parentDropdownGroupEl`

**Comment by benelan** [comment]
@driskull Is this issue still ready for verification after the revert in #5025?

**Comment by driskull** [comment]
Yes.

The navigation issue is because of the lack of a group.

**Comment by benelan** [comment]
Thanks. Closing this issue: every `dropdown-item` must have a parent `dropdown-group`. I put in a couple doc PRs to make that clear, including #5050
